server/api/rounds.py

```diff
@@ -120,6 +204,18 @@ def end_round(election: Election, round: Round):
 
 
 def is_round_complete(election: Election, round: Round) -> bool:
+    # For batch audits, check that all sampled batches have recorded results
+    if election.audit_type == AuditType.BATCH_COMPARISON:
+        num_batches_without_results: int = (
+            Batch.query.join(SampledBatchDraw)
+            .filter_by(round_id=round.id)
+            .outerjoin(BatchResult)
+            .group_by(Batch.id)
+            .having(func.count(BatchResult.batch_id) == 0)
```

Comment:
this count does the right thing of not counting null batch results in the outer join?

Reply:
well, i hope so! do the tests convince you that it does? if not, i'd like to add more tests